rust/rope/src/tree.rs

```diff
@@ -116,29 +123,48 @@ enum NodeVal<N: NodeInfo> {
 
 // also consider making Metric a newtype for usize, so type system can
 // help separate metrics
+
+/// A trait for quickly processing attributes of a NodeInfo.
+/// 
+/// For the conceptual background see the 
+/// [blog post, Rope science, part 2: metrics](https://github.com/google/xi-editor/blob/master/doc/rope_science/rope_science_02.md).
 pub trait Metric<N: NodeInfo> {
+    /// Return the number of boundarys in the NodeInfo::Leaf
+    /// 
+    /// TODO: What does the second argument do?
```

Comment:
To the best of my understanding, the second argument is the total size/length of the node, in base units.

Reply:
Why have two sources of truth, it is `N.len()` right?

Reply:
Good question. I'm also just reading through the code and trying to figure it out; might it be that only leaves have a `len`, but this is implemented on all nodes?

Reply:
On rereading the code, I think @Eh2406 is correct, the additional argument should be removed. I thought it might be more efficient to pass an argument rather than have the method reach into the `NodeInfo` struct, but on reflection there's no good reason for this.